### PR TITLE
Add 'version.txt' to project root and refactor automation to use it [ci full] [firefox-android: main]

### PR DESCRIPTION
**Pull request by ahal**
This is needed in order to add application-services to Shipit, which expects a `version.txt` file at the repository root. Rather than have this duplicated both here and `.buildconfig-android.yml`, this patch refactors automation to use `version.txt` instead.

Bug: 1842897

### Pull Request checklist ###

- **Breaking changes**:  This PR follows our [breaking change policy](https://github.com/mozilla/application-services/blob/main/docs/howtos/breaking-changes.md)
  - [x] This PR follows the breaking change policy:
     - This PR has no breaking API changes, or
     - There are corresponding PRs for our consumer applications that resolve the breaking changes and have been approved
- [x] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes a changelog entry in [CHANGELOG.md](../CHANGELOG.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [x] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[firefox-android: branch-name]` to the PR title.

### CODE REVIEW

File: settings.gradle [hunk 1]
```diff
@@ -66,6 +66,7 @@ if (file('local.properties').canRead()) {
 
 def calcVersion(buildconfig) {
     def local = gradle.rootProject.findProperty("local")
+    def version = new File(rootDir, 'version.txt').getText().trim()
```
Comment (ahal) [inline settings.gradle:69]: I'm not familiar with groovy.. When I went to put this assignment in the outer scope (where we read the `buildconfig`), it resulted in a failure.

File: docs/howtos/locally-published-components-in-fenix.md [hunk 2]
```diff
@@ -63,11 +63,11 @@ above. But if the auto-publishing workflow fails then it's important to know how
 on `android-components`, this procedure involves three separate repos:
 
 1. Inside the `application-services` repository root:
-    1. In [`.buildconfig-android.yml`](https://github.com/mozilla/application-services/blob/main/.buildconfig-android.yml), change
-       `libraryVersion` to end in `-TESTING$N` <sup><a href="#note1">1</a></sup>,
+    1. In [`version.txt`](https://github.com/mozilla/application-services/blob/main/version.txt), change
```
Comment (tarikeshaq) [inline docs/howtos/locally-published-components-in-fenix.md:66]: Thank you for changing the docs ❤️

**Review by tarikeshaq — APPROVED** [review]
Thanks @ahal! This looks great, and thanks a ton for taking the time to keep the docs up to date,

Before we land this, do you mind running `[ci full] [firefox-android: main]` on the PR title just as a check to make sure branch builds and all are ok?

Also one other thing to check if you haven't already is a local build of firefox-android, I can help run that test if you don't have the local environment ready!

Otherwise thanks a ton, this looks great!

**Comment by ahal** [comment]
> Before we land this, do you mind running [ci full] [firefox-android: main] on the PR title just as a check to make sure branch builds and all are ok?

Done, tasks are rerunning now.

> Also one other thing to check if you haven't already is a local build of firefox-android, I can help run that test if you don't have the local environment ready!

If you don't mind, I'd appreciate that! I don't have firefox-android setup.

**Comment by ahal** [comment]
Thanks @tarikeshaq! I'll deploy the related shipit PR as well, so we should be able to use shipit for the next release :)

/cc @bendk

**Pull request merged**